File: answers/leetcode/Product of Array Except Self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        product = 1
        product_list = []
        for i in range(len(nums)):
            product_list.append(product)
            product *= nums[i]
        product = 1
        for i in reversed(range(len(nums))):
            product_list[i] *= product
            product *= nums[i]
        return product_list

File: answers/leetcode/Product of Array Except Self/Product of Array Except Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] products = new int[nums.length];
        int product = 1;
        for (int i = 0; i < nums.length; i++) {
            products[i] = product;
            product *= nums[i];
        }
        product = 1;
        for (int i = nums.length - 1; i >= 0; i--) {
            products[i] *= product;
            product *= nums[i];
        }
        return products;
    }
}